libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -0,0 +1,29 @@
+use crate::semantic::convert_source;
+use crate::semantic::formatter::format;
+use pretty_assertions::assert_eq;
+
+#[test]
+fn format_semantic() {
+    let sem_pkg = convert_source(
+        r#"
+1 + 1
+n = 1.0
+f = n + 3.0
+s = "string"
+
+        "#,
+    )
+    .unwrap();
+
+    let got = format(&sem_pkg).unwrap();
+
+    // want
+    // 1 + 1  =>  1:int + 1:int
+    // n = 1.0  =>  n = 1.0:float
+    // x + 3.0  =>  “x:float + 3.0:float”
+    // s = “string” => s = “string”:string
+
+    let want = "package main\n1:int + 1:int\nn = 1.0:float\nf = n:float + 3.0:float\ns = \"string\":string";
```

Comment:
Can you use a raw string with `r#"..."` for `want` so that it looks pretty in the source code? When we write unit tests to assert on the structure of a semantic graph, I'd imagine that's what we'll do.

Reply:
Also, like we talked about I think it would be good to consider maybe omitting types where it's obvious (like in string literals). However I think we should always include the type for operators, since their type will always be inferred, as in `1:int +:int 1:int`.